roles/openshift_health_checker/openshift_checks/logging/podlogs.py

```diff
@@ -0,0 +1,174 @@
+        pods, error = super(PodLogQueryTime, self).get_pods_for_component(
+            self.execute_module,
+            self.logging_namespace,
+            "es",
+            task_vars,
+        )
+
+        if error:
+            return {"failed": True, "changed": False, "msg": error}
+
+        running_es_pods = self.running_pods(pods)
+        if not len(running_es_pods):
+            msg = ('No Elasticsearch pods were found to be in the "Running" state.'
+                   'At least one Elasticsearch pod is required in order to perform this check.')
+            return {"failed": True, "changed": False, "msg": msg}
+
+        error = self.check_log_delay(running_kibana_pods[0], running_es_pods[0], pod_log_timeout, task_vars)
+        if error:
+            return {"failed": True, "msg": error}
+
+        return {}
+
+    def check_log_delay(self, kibana_pod, es_pod, pod_log_timeout, task_vars):
+        """Check delay of logs and return any errors accordingly"""
+        error = None
+
+        try:
+            uid = self.curl_kibana_with_uuid(kibana_pod, task_vars)
+            self.wait_until_cmd_or_err(es_pod, uid, pod_log_timeout, task_vars)
+        except OpenShiftCheckException as error:
+            error = str(error)
+        except KeyError as error:
+            error = "Invalid json response from pod: {}".format(str(error))
+        except ValueError as error:
+            error = "Invalid response from pod {}".format(str(error))
+
+        return error
+
+    @staticmethod
+    def running_pods(pods):
+        """Returns: list of pods in a running state"""
+        return [
+            pod for pod in pods
+            if pod['status']['phase'] == 'Running'
+        ]
+
+    def wait_until_cmd_or_err(self, es_pod, uid, timeout_secs, task_vars):
+        """Wait a maximum of timeout_secs for the uuid logged in Kibana to be found in
+        the Elasticsearch logs. Since we are querying for a message with the the unique
+        uuid from earlier, there should only be a single match.
+        Raise an OpenShiftCheckException if the timeout is reached before a result is found."""
+
+        interval = 1  # seconds to wait between retries
+        orig_timeout = timeout_secs
+
+        while timeout_secs > 0:
+            time.sleep(interval)
+            timeout_secs -= interval
+
+            total = self.query_es_from_es(es_pod, uid, task_vars)
+            if total == 1:
+                return
+
+        msg = "expecting match in Elasticsearch for message with uuid {}, but no matches were found after {}s."
+        raise OpenShiftCheckException(msg.format(uid, orig_timeout))
+
+    def curl_kibana_with_uuid(self, kibanna_pod, task_vars):
+        """curl Kibana with a unique uuid."""
+        uid = str(self.generate_uuid())
+        pod_name = kibanna_pod["metadata"]["name"]
+        exec_cmd = "exec {pod_name} -c kibana -- curl --connect-timeout 60 -s http://localhost:5601/{uid}"
```

Comment:
It's a wild guess to some degree, but this is all intra-cluster networking. It would be a bad sign if connecting to a service IP took more than a second even if your cluster were federated around the world (although it could always be a transient network issue, I think it's much more likely to be something more significant). The actual retrieval of the request hits storage and processing so I could see a little more leeway on that, agree that it's a good idea to just use `--max-time` to cover the whole operation.